062-worker-pool/proposal.md

```diff
@@ -0,0 +1,157 @@
+# Summary
+
+This proposal outlines the support for worker pool. A worker pool is a group
+of Concourse workers, that can be associated with one or multiple teams.
+
+# Motivation
+
+One or multiple Concourse workers can be associated with a team by adding 
+`--team <team name>` option when launching `concourse worker`. Then all 
+pipeline containers of the team will be dispatched to the workers. The 
+relationship between workers and teams are N-to-1:
+
+```
+  Workers (0..n) <----------------> (0..1) Team
+```
+
+This is a useful mechanism to isolate some teams' workload or to provide
+dedicate workers to some teams.
```

Comment:
Maybe it's just that I run a (I guess, relatively) small Concourse instance, but I'm struggling to understand the underlying motivation here.

A core tenet of Concourse is avoiding worker state, with tags being a compromise to allow for e.g. hardened network architectures. Why do teams care about isolating their workload in the first place? How does one team's workload affect another team's workload? If it's to ensure worker availability for their projects, doesn't it make more sense to work on a feature like auto-scaling workers (i.e. "grow the pie") rather than a feature to protect and hoard "precious" workers (i.e. "divide the pie")?

If an organization *really* wants to provide a high level of isolation, why not set up isolated Concourse instances? Why only isolated Concourse worker pools?